### PR TITLE
Des 156 create the cambiar contrasena option from profile page phase 1 back end

### PR DESCRIPTION
**Pull request by Daniielmayo**

### CODE REVIEW

**Review by abdulflorez — DISMISSED** [review]
Looks good to me

File: controllers/auth.js [hunk 1]
```diff
@@ -192,6 +191,46 @@ const changePassword = async (req, res = response) => {
   }
 };
 
+const validatePassword = async (req, res) => {
+  const { id } = req.params;
+  const { password } = req.body;
+
+  try {
+    // Busca al usuario en la base de datos utilizando el ID
```
Comment (DanielCalleCO) [inline controllers/auth.js:199]: could you please translate this? Thanks buddy

File: controllers/auth.js [hunk 2]
```diff
@@ -192,6 +191,46 @@ const changePassword = async (req, res = response) => {
...
+    const user = await User.findById(id);
+
+    // Verifica si el usuario existe
```
Comment (DanielCalleCO) [inline controllers/auth.js:202]: translate it

File: controllers/auth.js [hunk 3]
```diff
@@ -192,6 +191,46 @@ const changePassword = async (req, res = response) => {
...
+    if (!user) {
+      return res.status(404).json({
+        ok: false,
+        msg: "Usuario no encontrado",
+      });
+    }
+
+    // Compara la contraseña proporcionada con la contraseña almacenada en la base de datos
```
Comment (DanielCalleCO) [inline controllers/auth.js:210]: translate this too

File: controllers/auth.js [hunk 4]
```diff
@@ -192,6 +191,46 @@ const changePassword = async (req, res = response) => {
...
+    const isPasswordValid = await bcrypt.compare(password, user.password);
+
+    if (!isPasswordValid) {
+      return res.status(400).json({
+        ok: false,
+        msg: "Contraseña incorrecta",
+      });
+    }
+
+    // Si la contraseña es válida, puedes responder con éxito
```
Comment (DanielCalleCO) [inline controllers/auth.js:220]: translate

File: controllers/auth.js [hunk 5]
```diff
@@ -192,6 +191,46 @@ const changePassword = async (req, res = response) => {
...
+    res.status(200).json({
+      ok: true,
+      msg: "Contraseña válida",
+    });
+  } catch (error) {
+    console.error("Error al validar la contraseña:", error);
+    res.status(500).json({
+      ok: false,
+      msg: "Por favor comuníquese con el administrador",
```
Comment (DanielCalleCO) [inline controllers/auth.js:229]:
```suggestion
      msg: "Por favor comunícate con el administrador",
```

**Review by DanielCalleCO — CHANGES_REQUESTED** [review]
Please translate the commented lines buddy and re send

**Review by DanielCalleCO — APPROVED** [review]

**Review by abdulflorez — APPROVED** [review]

**Pull request merged**